utils/get_data.py

```diff
@@ -1,26 +1,136 @@
 # -*- coding: utf-8 -*-
 """Module for getting the data from the Lizard API."""
 import json
+import time
 import urllib2
 import urllib
 
 from .constants import BASE_URL
+from .constants import ERROR_LEVEL_CRITICAL
+from .constants import ERROR_LEVEL_INFO
+from .user_communication import show_message
 
+DOWNLOAD_LIMIT = 1000
+PAYLOAD_FORMAT = "json"
+TASK_STATUS_PENDING = "PENDING"
+TASK_STATUS_SUCCESS = "SUCCESS"
+TASK_STATUS_FAILURE = "FAILURE"
+TASK_INTERVAL_MIN = 0.25
+TASK_INTERVAL_INCREMENT = 0.25
+TASK_INTERVAL_MAX = 5
 
-def get_data(asset_type, payload):
-    """Function to get the JSON from the Lizard client."""
-    # Encode the payload
-    payload_encoded = urllib.urlencode(payload)
 
-    # Set the url
-    url = "{}{}/?{}".format(
-        BASE_URL, asset_type,
-        payload_encoded)
+def get_data(username, password, asset_type, payload):
+    """
+    Function to get the JSON with asset data from the Lizard API.
+
+    username is the username used for logging into Lizard.
+    password is the password used for logging into Lizard.
+    """
+    # Download all the assets of an asset type
+    max_amount = get_max_amount(username, password, asset_type, payload)
+    payload["async"] = "true"
+    payload["format"] = PAYLOAD_FORMAT
+    payload["page_size"] = DOWNLOAD_LIMIT
+    payload["page"] = 1
+    current_amount = 0
+    results = []
+    queue_time_interval = TASK_INTERVAL_MIN
+
+    while current_amount < max_amount:
+        # Create a task to download the JSON async
+        task_start_url = "{}{}/".format(BASE_URL, asset_type)
+        task_start_json = get_json(username, password, task_start_url, payload)
+
+        # Get the task JSON
+        task_payload = {"format": "json"}
+        task_url = "{}".format(task_start_json["url"])
+        task_json = get_json(username, password, task_url, task_payload)
+
+        # Check whether the task is done
+        task_status = str(task_json["task_status"])
+        while task_status == TASK_STATUS_PENDING:
```

Comment:
This loop is probably not necessary. Without this loop you won't have to duplicate some of the code for task status.